### ISSUE TITLE
El Dorado: Only restrict zone to AVI,EXP for Reservations (instead of based on deliverydate)

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#1822]
_Imported from trac ticket 1822.
Created by **mariam**
Opened in trac: Jul 23, 2012, 5:18:21 PM
Last modified in trac: Jul 24, 2012, 11:59:42 AM_
***

**Comment by kbower** [comment]
_**kbower** trac comment on Jul 24, 2012, 8:52:18 AM:_

The rule part already exists:
```python
if order.status == u'V' or order._original_status_ != u'O':
    return
if order.type == u'RES':
    if order.deliverydate not in (DeferredStatus.ELD_UNSCHEDULED, DeferredStatus.ELD_UNSCHEDULED_EXPORT):
        order.error('Delivery date must be Unscheduled for Reservations','deliverydate')
    if order.shipzoneid not in (u'AVI',u'EXP'):
        order.error('Ship zone must be AVI (or export) for Reservations','shipzoneid')
```_Note that the rule is needed to prevent someone playing with the order type drop down in order to temporarily change the type to Sale just long enough to pick a "real" zone_

**Comment by kbower** [comment]
_**kbower** trac comment on Jul 24, 2012, 11:59:42 AM:_

(Changeset [[2914]](../commit/c8ccbc1fa60ffa9d4316971b9bb8dd91cb771ac0))
 * be more conservative about defaulting the fromsiteid for pickups (requiring user input because of tax implications)
 * close #1822 El Dorado: Only restrict zone to AVI,EXP for Reservations (instead of based on deliverydate)
 * fix [#1823](1823) El Dorado: when deliverytype changed to 'P', blank fromsiteid (fix to not set to ordersiteid at save time)

**Comment by kbower** [comment]
_**kbower** trac status change on Jul 24, 2012, 11:59:42 AM:_ **closed**
